src/Dofs/ConstraintHandler.jl

```diff
@@ -61,6 +61,78 @@ function __to_components(c)
     return components
 end
 
+@doc raw"""
+    WeakDirichlet(field_name::Symbol, facets::AbstractSet{FacetIndex}, f::Function; qr_order = -1)
```

Comment:
Can we discuss the name for a moment? I think `WeakDirichlet` is misleading, because at least from the literature I know ,,weak bcs'' are usually introduced by adding energy terms (or weak form terms if we have no energy) to force the solution towards the right boundary condition, as done e.g. with Nitzsche method. However, the presented method here strongly enforces the boundary condition.

Can we merge this infrastructure into the `Dirichlet` struct directly and control the application by dispatching on the new conformity trait?

I also noticed that we are missing some docstring for the application of DBCs in L2 elements. :D

Reply:
> Can we merge this infrastructure into the Dirichlet struct directly and control the application by dispatching on the new conformity trait?

I think it would be wiser not to do that (ref: https://github.com/Ferrite-FEM/Ferrite.jl/pull/1151#issuecomment-2665862101 above). But of course, we could consider changing `Dirichlet` to be a wrapper/input type, and then store separate structs internally depending on how we want to apply BCs.

Reply:
> However, the presented method here strongly enforces the boundary condition.

You are right, ref [Bazilevs and Hughes (2007) Weak imposition of Dirichlet boundary conditions in fluid mechanics](https://www.sciencedirect.com/science/article/pii/S0045793005001258)

However, if the method here is used for an H1 interpolation, it would give a different result compared to the current Dirichlet which satisfies the prescribed value point-wise, whereas the present one only in a weak/integral sense.

How about going back to the original name - "IntegrableDirichlet"?

Reply:
> However, if the method here is used for an H1 interpolation, it would give a different result compared to the current Dirichlet which satisfies the prescribed value point-wise, whereas the present one only in a weak/integral sense.

Oh, i see, it "smears out" the Boundary condition because we use Gauss-Legendre quadrature by default, right? The method should be exact when using Gauss-Lobatto quadrature tho.

Reply:
Yes, assuming the Gauss-Lobatto has equally many points as the number of dofs, but with higher integration orders it doesn't matter what integration rule is used. I think of it like this: 

![image](https://github.com/user-attachments/assets/8d550d89-e33e-41a0-a04e-a6e00c74fa8c)

Reply:
This is indeed quite interesting, I have not thought about non-matching functions here. So the new boundary condition enforcement should be closer to the actual function on our Dirichlet boundary, because the new method minimizes the distance between the analytical function and our function space, while the standard method is "just" interpolatory.

Reply:
I think we should add this sketch to the docs as it very precisely explains the difference to users.

Reply:
https://ferrite-fem.github.io/Ferrite.jl/previews/PR1151/topics/boundary_conditions/#L2ProjectedDirichlet